tests/utils/helpers.ts

```diff
@@ -0,0 +1,244 @@
+   expect(slider).not.toBeNull();
+
+   if (slider) {
+      // Get slider data
+      const dataBefore: FilterData = JSON.parse(slider.getAttribute('data-test-option') as string);
+      expect(dataBefore.value).toBe(0);
+      // Change value of slider
+      fireEvent.change(slider, { target: { value: randomValue } });
+      // Get data after change and compare it with previous data
+      const dataAfter: FilterData = JSON.parse(slider.getAttribute('data-test-option') as string);
+      // Data value before and after should be different
+      expect(dataAfter.value).toBe(randomValue);
+
+      // Check if offers are correctly filtered
+      expect(await isOfferListFiltered(dataAfter)).toBeTruthy();
+      // Reset slider to default state
+      fireEvent.change(slider, { target: { value: 0 } });
+      const dataReset: FilterData = JSON.parse(slider.getAttribute('data-test-option') as string);
+      expect(dataReset.value).toBe(0);
+   }
+};
+
+export const testFilterTextInput = async (name: string) => {
+   const textInput = screen.getByTestId(`search-${name}`) as HTMLInputElement;
+   const dataBefore: FilterData = JSON.parse(textInput.getAttribute('data-test-option') as string);
+   const valueBefore = textInput.value;
+   // Init value of text input should be empty string
+   expect(valueBefore).toBe('');
+   expect(dataBefore.value).toBe('');
+
+   // Click on text input and type sth
+   // (eg. for location "ne" should display offers in Sydney and New York)
+   // (eg. for job title "ne" should display offers like "Software Engineer" "Business Analyst" etc.)
+   await userEvent.click(textInput);
+   const testText = 'ne';
+   await userEvent.keyboard(testText);
+
+   // Check if input value is equal to provided value
+   const dataAfter: FilterData = JSON.parse(textInput.getAttribute('data-test-option') as string);
+   const valueAfter = textInput.value;
+   expect(valueAfter).toBe(testText);
+   expect(dataAfter.value).toBe(testText);
+
+   // Check if offers are correctly filtered
+   expect(await isOfferListFiltered(dataAfter)).toBeTruthy();
+
+   // Set the text input value to an empty string
+   await userEvent.clear(textInput);
+   // Verify if the value is now an empty string
+   const dataReset: FilterData = JSON.parse(textInput.getAttribute('data-test-option') as string);
+   expect(textInput.value).toBe('');
+   expect(dataReset.value).toBe('');
+};
+
+// Check if offer list is filtered according to setted filter
+const isOfferListFiltered = async (filter: FilterData): Promise<boolean> => {
+   // Get currently displayed offer list
+   const offers = await getOffersData();
+   // Loop over offers and check if there are filtered offers
+   if (offers.length > 0) {
+      for (const offer of offers) {
+         if (typeof offer[filter.offerKeyName] === 'number' && typeof filter.value === 'number') {
+            // Compare number values
+            if (filter.offerOption.includes('-min')) {
```

Comment:
**nitpick:** This is very complicated, and unreadable - I would split it explicitly for each filter, so it's clear what's going on. Eg. `isJobTypeSelected`